src/text-field.jsx

```diff
@@ -331,7 +331,8 @@ const TextField = React.createClass({
       inputProps.onChange = this._handleInputChange;
     }
     if (this.props.children) {
-      inputElement = React.cloneElement(this.props.children, {...inputProps, ...this.props.children.props});
+      let childInputStyle = this.mergeStyles(inputStyle, this.props.children.style);
```

Comment:
Could you remove the extra commit from https://github.com/callemall/material-ui/pull/1896?